functions/supplier/sftp-supplier/src/main/java/org/springframework/cloud/fn/supplier/sftp/SftpSupplierFactoryConfiguration.java

```diff
@@ -84,9 +85,10 @@ static SessionFactory<SftpClient.DirEntry> buildFactory(BeanFactory beanFactory,
 		sftpSessionFactory.setPrivateKeyPassphrase(factory.getPassPhrase());
 		sftpSessionFactory.setAllowUnknownKeys(factory.isAllowUnknownKeys());
 		if (factory.getKnownHostsExpression() != null) {
-			String path = factory.getKnownHostsExpression()
+			String knownHostsLocation = factory.getKnownHostsExpression()
 					.getValue(IntegrationContextUtils.getEvaluationContext(beanFactory), String.class);
-			sftpSessionFactory.setKnownHostsResource(new FileSystemResource(path));
+			Resource knownHostsResource = new DefaultResourceLoader().getResource(knownHostsLocation);
```

Comment:
Any reason not to use the SpEL method param to get the resource like you are doing above for `SftpConsumerSessionFactoryConfiguration`?  I am ok w/ either approach but for consistency, can they be the same?

Also, if we stick w/ the programmatic route, wdyt about using the container's `ResourceLoader` (either via method param injection or via ResourceLoaderAware)?